examples/postgresql/README.md

```diff
@@ -0,0 +1,109 @@
+
+
+## Guide
+
+### Considerations
+The following assumptions are made:
++ At least one SPIRE server and one agent are deployed with trust domain `example.org`.
+
++ Postgres 12 is used in this guide but each step should be easily applicable to other versions as well.
+
+### 1. Install PosgreSQL
+Install [PosgreSQL](https://www.postgresql.org/docs/12/tutorial-install.html) and make sure the service is up running.
+```
+systemctl status postgresql@12-main
+```
+
+### 2. Create the user
+Create the `postgres-user` using the [provided script](create_user.sql).
+```sql
+sudo -u postgres psql -c create_user.sql
+```
+
+### 3. Configure PosgreSQL SSL settings
+Make PostgreSQL server to use the certificates and key provided by SPIRE by setting the following SSL configurables in the PostgreSQL configuration file (`postgresql.conf`).
+```bash
+ssl = on
+ssl_cert_file = '/opt/spire/certs/postgresql/svid.pem'
+ssl_key_file = '/opt/spire/certs/postgresql/svid.key'
+ssl_ca_file = '/opt/spire/certs/postgresql/svid_bundle.pem'
+```
+
+Create the folder using the `postgres` user:
+```
+sudo -u postgres mkdir -p /opt/spire/certs/postgresql
+```
+
+### 4. Configure pg_hba.conf
+Create the following rules in the `pg_hba.conf` file:
+```
+# TYPE      DATABASE        USER            ADDRESS                 METHOD
+
+# Allows postgres user to connect using unix domain sockets (not required)
+local       all             postgres                                trust
+
+# Reject all nossl calls (not required)
+hostnossl   all             all             0.0.0.0/0               reject
+
+# Validate provided certificates
+hostssl     all             all             0.0.0.0/0               cert clientcert=1
+```
+
+### 5. Start SPIRE server
+Start the SPIRE server:
+```bash
+./spire-server run
+```
+
+### 6. Start SPIRE agent
+Start the SPIRE agent:
+```bash
+./spire-server bundle show > conf/agent/dummy_root_ca.crt
+TOKEN=$(./spire-server token generate -spiffeID spiffe://example.org/agent)| awk '{print $2}')
+./spire-agent run -joinToken $TOKEN
+```
+
+### 7. Create the registration entries
+Create the following registration entries:
+
++ For the PostgreSQL client, the DNS name must match the database user name. The selector used in this case is your current user id.
+```bash
+./spire-server entry create \
+    -spiffeID spiffe://example.org/psql-client \
+    -parentID spiffe://example.org/agent \
+    -selector unix:uid:1000 \
+    -ttl 60 \
+    -dns postgres-user
+```
+
++ For the PostgreSQL server, we use the postgres user name as selector:
+```bash
+./spire-server entry create \
+    -spiffeID spiffe://example.org/postgresql-server \
+    -parentID spiffe://example.org/agent \
+    -selector unix:user:postgres \
+    -ttl 60
+```
+
+Note that `ttl` is lowered to 60 seconds in both cases just for demo purposes.
+
+
+### 8. Start spiffe-helper
+Start spiffe-helper using this example [configuration file](examples/postgresql/helper.conf) with the `postgres` user:
```

Comment:
can you explain what it is executed every time a new certificate is found? or at least mention that we are reloading certificates every time it rotate?